lcb_runner/runner/base_runner.py

```diff
@@ -96,9 +97,37 @@ def run_batch(self, prompts: list[str | list[dict[str, str]]]) -> list[str]:
         return outputs
 
     def run_main(self, benchmark: list, format_prompt: callable) -> list:
-        prompts = [
-            format_prompt(problem, self.model.model_style) for problem in benchmark
-        ]
+        if self.args.scenario == Scenario.selfrepair:
+            with open(f"output/{self.model.model_repr}/Scenario.codegeneration_10_{self.args.temperature}_eval_all.json") as f:
+                check_metadata = json.load(f)
+            outputs = []
+            for check in tqdm(check_metadata):
+                output = []
+                checked_base_question_cotent = check["question_content"]
+                checked_base_codes = check["code_list"]
+                checked_output_codes = check["output_list"]
+                checked_base_results = check["graded_list"]
+                checked_base_metadata = check["metadata"]
+                for i in range(len(checked_base_codes)):
+                    prompt=(
+                        format_prompt(
+                            checked_base_question_cotent,
+                            self.model.model_style,
+                            checked_base_codes[i],
+                            checked_base_results[i],
+                            checked_base_metadata[i],
+                        )
+                    )
+                    if prompt == "" or type(prompt) is not list:
```

Comment:
what is the second condition?

Reply:
I think the second condition is unnecessary since only empty string means that the previous generation succeeds